gateway/src/test/java/org/georchestra/gateway/security/ResolveGeorchestraUserGlobalFilterTest.java

```diff
@@ -115,7 +115,7 @@ void testFilter_UseResolved() {
 
         filter.filter(exchange, mockChain).block();
 
-        verify(mockChain, times(1)).filter(same(exchange));
+        verify(mockChain, times(2)).filter(same(exchange));
```

Comment:
Not much satisfied having to change this, as this test can no longer guarantee that `exchange.filter` won't be called twice, even though it is not possible in tested method.

Reply:
OK I messed up some things between my local version and this one, here `chain.filter(exchange)` is really called twice, which is wrong. So I need to find a way to replace `.defaultIfEmpty(chain.filter(exchange))`  in `gateway/src/main/java/org/georchestra/gateway/security/ResolveGeorchestraUserGlobalFilter.java` by something which calls `chain.filter(exchange)` only if it receives an empty stream.

I tried to replace it with `.switchIfEmpty(Mono.fromRunnable(() -> chain.filter(exchange)))` which satisfies the test but then I cannot access any service (like `/datahub/`) when I'm not logged in, just get an empty 200 response.